src/RequestBuilder/AbstractRequestBuilder.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\Matej\RequestBuilder;
+
+use Lmc\Matej\Exception\LogicException;
+use Lmc\Matej\Http\RequestManager;
+use Lmc\Matej\Model\Request;
+use Lmc\Matej\Model\Response;
+
+/**
+ * Request builders provides methods for simple and type-safe assembly of request to specific Matej endpoint.
+ *
+ * If `RequestManager` is injected to the builder via `setRequestManager()`, the request could be executed right from
+ * the builder using `send()` method.
+ */
+abstract class AbstractRequestBuilder
+{
+    /** @var RequestManager */
+    protected $requestManager;
+
+    /**
+     * Use Commands and other settings which were passed to this builder object to build instance of
+     * Lmc\Matej\Model\Request.
+     */
+    abstract public function build(): Request;
+
+    /**
+     * If instance of RequestManager is injected to this builder object, you can build and send the request directly
+     * via send() method of the builder itself.
+     */
+    public function setRequestManager(RequestManager $requestManager): self
+    {
+        $this->requestManager = $requestManager;
+
+        return $this;
+    }
+
+    public function send(): Response
+    {
+        $this->assertRequestManagerIsAvailable();
+
+        return $this->requestManager->sendRequest($this->build());
+    }
+
+    private function assertRequestManagerIsAvailable(): void
+    {
+        if ($this->requestManager === null) {
```

Comment:
How do you feel about `if (!$this->requestManager instanceof RequestManager)` ?

Reply:
I don't feel much difference here. Maybe prefer this version a little bit, as I read it like "there is no requeste manager", while the instanceof version is more like "request manager is request manager". 

What about you :)? Any advantage for the instanceof version?